docs/DeveloperGuide.md

```diff
@@ -1,38 +1,324 @@
+    * [Storage Component](#storage)
+    * [Validator Component](#validator)
+  * [Asset Classes](#asset-classes)
 
-{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+* [Implementation](#implementation)
+  * [Design Considerations](#design-considerations)
+* [Product Scope](#product-scope)
+  * [Target User](#target-user-profile)
+  * [Value Proposition](#value-proposition)
+  * [User Stories](#user-stories)
+  * [Use Cases](#use-cases)
+  * [Non-Functional Requirements](#non-functional-requirements)
+* [Glossary](#glossary)
+* [Appendix](#appendix)
 
-## Design & implementation
+--------------------------------------------------------------------------------------------------------------------
+## **Acknowledgements**
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+* Design and Structure of Developer Guide referenced from the
+[AB3-Developer Guide](http://se-education.org/addressbook-level3/DeveloperGuide.html).
 
+--------------------------------------------------------------------------------------------------------------------
+## **Getting Started for Beginners**
 
+Refer to the [_User Guide_](UserGuide.md).
+
+--------------------------------------------------------------------------------------------------------------------
+## **Design**
+
+### Architecture
+
+![Architecture Diagram](diagrams/OverallArch.png)
+
+The ***Architecture Diagram*** given above explains the high-level design ***HalpMi***.
+
+Given below is a quick overview of main components and how they interact with each other.
+
+#### **Main components of the architecture**
+
+**`Duke`** has single method called `main` which is called upon launch. This initialises a new instance of a `Manager` class, and calls the `runLoop()`
+method belonging to the Manager object.
+
+[**`Assets`**](#asset-classes): Refers to a collection of classes that hold all the necessary data given by the User in
+current and past usages.
+
+The rest of the App consists of these components.
+
+* [**`Manager`**](#manager-component): The Brain.
+* [**`Helper`**](#helper-classes): A collection of core classes that aid with the Logical Operations performed by HalpMi.
+    * [**`Command`**](#command): Contains the changes or updates to be made.
+    * [**`UI`**](#ui-component): The UI of the App.
+    * [**`Parser`**](#parser): Breaks down user input into parameters accepted by the app and creates a Command Object.
+    * [**`Validator`**](): Checks if the input provided by the User is Valid.
+    * [**`Storage`**](#storage): Reads data from data files and writes data to data files, also stores in app memory.
+
+The Sequence Diagram below showcases the general Logic and Flow of the program from Launch till Exit.
+
+![Sequence Diagram](diagrams/SequenceDiagram.png)
+
+### UI component
+
+![UIClassUML](diagrams/UIClassUML.png)
```

Comment:
Consider following the lecture uml diagram notation for showing visibility and abstract classes.